core/GlobalState.cc

```diff
@@ -2160,7 +2161,11 @@ unique_ptr<LocalSymbolTableHashes> GlobalState::hash() const {
         counter++;
         // No fields are ignored in hashing.
         uint32_t symhash = field.hash(*this);
-        hierarchyHash = mix(hierarchyHash, symhash);
+        if (field.flags.isStaticField) {
+            auto &target = staticFieldHashesMap[NameHash(*this, field.name)];
+            target = mix(target, symhash);
+        }
+        hierarchyHash = mix(hierarchyHash, field.fieldShapeHash(*this));
```

Comment:
Do the changes in this PR work the same for fields like instance variables as well? (Is support for instance variables as easy as making this section unconditional?)

Reply:
fwiw, it would be really cool if we could get instance variable working, because it would mean that changes to types of `prop`'s and `attr_*`'s would take the fast path.